pkg/controller/replication_controller_test.go

```diff
@@ -41,31 +43,61 @@ type FakePodControl struct {
 	controllerSpec []api.ReplicationController
 	deletePodName  []string
 	lock           sync.Mutex
+	err            error
 }
 
 func init() {
 	api.ForTesting_ReferencesAllowBlankSelfLinks = true
 }
 
-func (f *FakePodControl) createReplica(namespace string, spec api.ReplicationController) {
+func (f *FakePodControl) createReplica(namespace string, spec api.ReplicationController) error {
 	f.lock.Lock()
 	defer f.lock.Unlock()
+	if f.err != nil {
+		return f.err
+	}
 	f.controllerSpec = append(f.controllerSpec, spec)
+	return nil
 }
 
 func (f *FakePodControl) deletePod(namespace string, podName string) error {
 	f.lock.Lock()
 	defer f.lock.Unlock()
+	if f.err != nil {
+		return f.err
+	}
 	f.deletePodName = append(f.deletePodName, podName)
 	return nil
 }
 
-func newReplicationController(replicas int) api.ReplicationController {
-	return api.ReplicationController{
-		TypeMeta:   api.TypeMeta{APIVersion: testapi.Version()},
-		ObjectMeta: api.ObjectMeta{Name: "foobar", Namespace: api.NamespaceDefault, ResourceVersion: "18"},
+func (f *FakePodControl) clear() {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+	f.deletePodName = []string{}
+	f.controllerSpec = []api.ReplicationController{}
+}
+
+func getKey(rc *api.ReplicationController, t *testing.T) string {
+	if key, err := rcKeyFunc(rc); err != nil {
+		t.Errorf("Unexpected error getting key for rc %v: %v", rc.Name, err)
+		return ""
+	} else {
+		return key
+	}
+}
+
+func newReplicationController(store cache.Store, replicas int) *api.ReplicationController {
```

Comment:
Can, I was trying to follow a pattern with newPodList which also takes a store and doing that in the caller is bound to be repetative since it returns a list, but rcs alone should be OK.